tokio/src/sync/watch.rs

```diff
@@ -23,8 +21,8 @@
 //!     let (tx, mut rx) = watch::channel("hello");
 //!
 //!     tokio::spawn(async move {
-//!         while let Some(value) = Some(rx.recv().await) {
-//!             println!("received = {:?}", value);
+//!         while rx.changed().await.is_ok() {
+//!             println!("received = {:?}", *rx.borrow());
 //!         }
 //!     });
```

Comment:
One possible API thought: what if `changed()` returned a thing that you could call `borrow` on? Then you could write
```rust
tokio::spawn(async move {
    println!("current state = {:?}", *rx.borrow());
    while let Ok(change) = rx.changed().await {
        println!("received = {:?}", *change.borrow());
    }
});
```

I'm not sure if this is useful enough to be worth the effort...

Reply:
Borrowing and waiting for a change is pretty decoupled. I don't think there is any performance gain to batching the operations. I opted for keeping the API simpler.